templates/bionic/etc/fail2ban/jail.conf.epp

```diff
@@ -0,0 +1,984 @@
+logpath = /var/log/ejabberd/ejabberd.log
+
+
+[counter-strike]
+
+enabled = <%= "counter-strike" in $fail2ban::jails %>
+logpath = /opt/cstrike/logs/L[0-9]*.log
+# Firewall: http://www.cstrike-planet.com/faq/6
+tcpport = 27030,27031,27032,27033,27034,27035,27036,27037,27038,27039
+udpport = 1200,27000,27001,27002,27003,27004,27005,27006,27007,27008,27009,27010,27011,27012,27013,27014,27015
+action  = %(banaction)s[name=%(__name__)s-tcp, port="%(tcpport)s", protocol="tcp", chain="%(chain)s", actname=%(banaction)s-tcp]
+           %(banaction)s[name=%(__name__)s-udp, port="%(udpport)s", protocol="udp", chain="%(chain)s", actname=%(banaction)s-udp]
+
+# consider low maxretry and a long bantime
+# nobody except your own Nagios server should ever probe nrpe
+[nagios]
+
+enabled  = <%= "nagios" in $fail2ban::jails %>
+logpath  = %(syslog_daemon)s     ; nrpe.cfg may define a different log_facility
+backend  = %(syslog_backend)s
+maxretry = 1
+
+
+[oracleims]
+# see "oracleims" filter file for configuration requirement for Oracle IMS v6 and above
+enabled = <%= "oracleims" in $fail2ban::jails %>
+logpath = /opt/sun/comms/messaging64/log/mail.log_current
+banaction = %(banaction_allports)s
+
+[directadmin]
+
+enabled = <%= "directadmin" in $fail2ban::jails %>
+logpath = /var/log/directadmin/login.log
+port = 2222
+
+[portsentry]
+
+enabled  = <%= "portsentry" in $fail2ban::jails %>
+logpath  = /var/lib/portsentry/portsentry.history
+maxretry = 1
+
+[pass2allow-ftp]
+# this pass2allow example allows FTP traffic after successful HTTP authentication
+enabled  = <%= "pass2allow-ftp" in $fail2ban::jails %>
+port         = ftp,ftp-data,ftps,ftps-data
+# knocking_url variable must be overridden to some secret value in jail.local
+knocking_url = /knocking/
+filter       = apache-pass[knocking_url="%(knocking_url)s"]
+# access log of the website with HTTP auth
+logpath      = %(apache_access_log)s
+blocktype    = RETURN
+returntype   = DROP
+action       = %(action_)s[blocktype=%(blocktype)s, returntype=%(returntype)s]
+bantime      = 1h
+maxretry     = 1
+findtime     = 1
+
+
+[murmur]
+# AKA mumble-server
+enabled  = <%= "murmur" in $fail2ban::jails %>
+port     = 64738
+action   = %(banaction)s[name=%(__name__)s-tcp, port="%(port)s", protocol=tcp, chain="%(chain)s", actname=%(banaction)s-tcp]
+           %(banaction)s[name=%(__name__)s-udp, port="%(port)s", protocol=udp, chain="%(chain)s", actname=%(banaction)s-udp]
+logpath  = /var/log/mumble-server/mumble-server.log
+
+
+[screensharingd]
+# For Mac OS Screen Sharing Service (VNC)
+enabled  = <%= "screensharingd" in $fail2ban::jails %>
+logpath  = /var/log/system.log
+logencoding = utf-8
+
+[haproxy-http-auth]
+# HAProxy by default doesn't log to file you'll need to set it up to forward
+# logs to a syslog server which would then write them to disk.
+# See "haproxy-http-auth" filter for a brief cautionary note when setting
+# maxretry and findtime.
+enabled  = <%= "haproxy-http-auth" in $fail2ban::jails %>
```

Comment:
oh this syntax is pretty neat. Does it return true/false as a string? Can you add a test for this?

Reply:
It's an epp template, so it uses the puppet `in` comparison operator as shown here: https://puppet.com/docs/puppet/6.1/lang_expressions.html#in
It resolves to true/false.

Also, it's not new in this template, as it's present in all the previous ones.

I think it's already tested here
https://github.com/voxpupuli/puppet-fail2ban/blob/e2a76a96a5cecdd3977bd566b904966832b5609b/spec/acceptance/class_spec.rb#L237-L244
since by default ssh and ssh-ddos jails are enabled, and that syntax is used to configure those jails. E.g.:
https://github.com/voxpupuli/puppet-fail2ban/blob/e2a76a96a5cecdd3977bd566b904966832b5609b/templates/xenial/etc/fail2ban/jail.conf.epp#L223-L228